app/vmalert/config/config.go

```diff
@@ -202,9 +215,27 @@ func parseFile(path string) ([]Group, error) {
 	if err != nil {
 		return nil, err
 	}
+	g.Groups = applyGlobal(g.Groups, g.Global)
```

Comment:
the caller ([func Parse](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/5b0925d07d448e752d8889b2045eb21012270785/app/vmalert/config/config.go#L184)) already iterates over all groups. I think it would be easier to add a one line there instead of creating a new function.